tests/e2e/tests/drafts/drafts.js

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const test = require('ava');
+const fs = require('fs');
+const path = require('path');
+
+// eslint-disable-next-line ava/no-import-test-files
+const {distPath} = require('../../utils.js');
+
+test('Output a non-draft file', t => {
+  const filePath = path.join(distPath, 'en/drafts/not-a-draft/index.html');
+  const statsObj = fs.statSync(filePath);
+
+  t.truthy(statsObj);
+});
+
+test('Outputs a past file', t => {
+  const filePath = path.join(distPath, 'en/drafts/past-post/index.html');
+  const statsObj = fs.statSync(filePath);
+
+  t.truthy(statsObj);
+});
+
+test('Outputs a past file with canonical date', t => {
+  const filePath = path.join(
+    distPath,
+    'en/drafts/past-post-date-canonical/index.html'
+  );
+  const statsObj = fs.statSync(filePath);
+
+  t.truthy(statsObj);
+});
+
+test('Outputs a past file with date with space', t => {
+  const filePath = path.join(
+    distPath,
+    'en/drafts/past-post-date-with-space/index.html'
+  );
+  const statsObj = fs.statSync(filePath);
+
+  t.truthy(statsObj);
+});
+
+test('Outputs a past file with date with timezone', t => {
+  const filePath = path.join(
+    distPath,
+    'en/drafts/past-post-date-with-timezone/index.html'
+  );
+  const statsObj = fs.statSync(filePath);
+
+  t.truthy(statsObj);
+});
+
+if (process.env.NODE_ENV === 'production') {
```

Comment:
I can't wrap my head around this: when would this be true in the test environment?

Reply:
Yeah, it's not ideal. i'm still a bit unsure about this approach myself. It will never trigger on automated tests, but you can trigger it manually. Since the drafts behave differently in prod and non-prod by design, we should find a way to test the difference. 
I wonder if maybe all e2e tests should assume prod. Or we introduce test as a separate environment and differentiate behavior based on that...

Reply:
As discussed in chat: prod for all e2e tests sound reasonable